compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1101,6 +1123,10 @@ public void visit(BLangSimpleVariableDef astVarDefStmt) {
             this.currentScope = newScope;
         }
 
+        if (astVarDefStmt.var.typeNode != null) {
+            astVarDefStmt.var.typeNode.accept(this);
```

Comment:
Extract `astVarDefStmt.var.typeNode` out to a variable.

Reply:
Please update wherever applicable.

Reply:
Changed the logic https://github.com/ballerina-platform/ballerina-lang/pull/39101/commits/ed9d56f60b01a2744651c7e805361dd574b8357d